core/src/main/resources/application.conf

```diff
@@ -84,6 +84,10 @@ system {
 
   // Since the WorkflowLogCopyRouter is initialized in code, this is the number of workers
   number-of-workflow-log-copy-workers = 10
+
+  // Whether to run the metadata summarizer in this Cromwell instance.  There should only be one metadata summarizer
+  // per database, so if this Cromwell instance is being used in read-only mode this value should be `false`.
```

Comment:
I'd rather see a few general purposes switches that we can figure out the
right way to take advantage of at this time, than specialized
roles/behaviors that we're not sure if they're useful.

the special flag here is ok -- I had mentioned in the ticket just reusing
the refresh interval since intuitively setting that to infinity/zero says
to me that it won't happen and doesn't introduce a special flag... but
they're close enough in intent that i'm fine with either (esp since the
special flag is already done).



-------------------------------
Kristian Cibulskis
Chief Architect, Data Sciences & Data Engineering
Broad Institute of MIT and Harvard
kcibul@broadinstitute.org



On Fri, Sep 9, 2016 at 1:41 PM, Chris Llanwarne <notifications@github.com>
wrote:

> In core/src/main/resources/application.conf
> <https://github.com/broadinstitute/cromwell/pull/1385#discussion_r78219549>
> :
>
> > @@ -84,6 +84,10 @@ system {
> >
> >    // Since the WorkflowLogCopyRouter is initialized in code, this is the number of workers
> >    number-of-workflow-log-copy-workers = 10
> > +
> > +  // Whether to run the metadata summarizer in this Cromwell instance.  There should only be one metadata summarizer
> > +  // per database, so if this Cromwell instance is being used in read-only mode this value should be `false`.
>
> Possible Cromwell roles:
>
>    - Executor
>    - Metadata Reader
>    - Database Maintainer (i.e. summariser)
>    - Submitter
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/broadinstitute/cromwell/pull/1385/files/37e8a49d1bf7dd628e6035bb9e279bb3d31d922f#r78219549>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABW4g2S56SzXXs9Kjlrau0ZwrwznBF9nks5qoZougaJpZM4J5PYi>
> .
>